pkg/dao/repository_config.go

```diff
@@ -322,6 +324,74 @@ func ModelToApiFields(repoConfig models.RepositoryConfiguration, apiRepo *api.Re
 	}
 }
 
+func convertSortByToSQL(SortBy string) string {
+	sqlOrderBy := ""
+
+	sortByArray := strings.Split(SortBy, ",")
+	lengthOfSortByParams := len(sortByArray)
+
+	for i := 0; i < lengthOfSortByParams; i++ {
+		order := ""
+		switch sortByArray[i] {
+		case "name":
+			fallthrough
+		case "name:asc":
+			order = "name asc"
+		case "name:desc":
+			order = "name desc"
+
+		case "url":
+			fallthrough
+		case "url:asc":
+			order = "url asc"
+		case "url:desc":
+			order = "url desc"
+
+		case "distribution_arch":
+			fallthrough
+		case "distribution_arch:asc":
+			order = "arch asc"
+		case "distribution_arch:desc":
+			order = "arch desc"
+
+		case "distribution_versions":
+			fallthrough
+		case "distribution_versions:asc":
+			order = "array_to_string(versions, ',') asc"
+		case "distribution_versions:desc":
+			order = "array_to_string(versions, ',') desc"
+
+		case "package_count":
+			fallthrough
+		case "package_count:asc":
+			order = "package_count asc"
+		case "package_count:desc":
+			order = "package_count desc"
+
+		case "status":
+			fallthrough
+		case "status:asc":
+			order = "status asc"
+		case "status:desc":
+			order = "status desc"
+		}
```

Comment:
This is a tad verbose, what about something like this:

	sortBy := sortByArray[i]

	split := strings.Split(sortBy, ":")
	if len(split) < 2 {
		order = "asc"
	} else {
                order = split[1]
         }

        if order != "asc" && order != "desc" {
                //Throw error?
         }

	n := map[string]string{
		"name":                  "name",
		"url":                   "url",
		"distribution_arch":     "distribution_arch",
		"distribution_versions": "array_to_string(versions, ',')",
		"package_count":         "package_count",
		"status":                "status",
	}

	orderField, ok := n[split[0]]
	if !ok {
		//throw error?
	}
	sqlOrderBy = orderField + " " + split[1]

Reply:
Oh I see what you mean here, just cleaning up the switch statement a bit.

Reply:
I see what you've done here with creating that array. 
But as we are already A. Looping and B. concatenating strings, I'd prefer not to prepare another loop for clarity.
I do very much like what you have done with the map, though, and have incorporated that change.

Reply:
iI can go ahead with adding the looping method if you'd strongly prefer it. O^n+1 is probably fine.